tests/api2/test_108_catalog.py

```diff
@@ -0,0 +1,187 @@
+
+import os
+import pytest
+import sys
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from functions import GET, POST, DELETE, PUT, wait_on_job
+from auto_config import ha, scale, dev_test
+
+if dev_test:
+    reason = 'Skip for testing'
+else:
+    reason = 'Skipping test for HA' if ha else 'Skipping test for CORE'
+# comment pytestmark for development testing with --dev-test
+pytestmark = pytest.mark.skipif(ha or not scale or dev_test, reason=reason)
+
+official_repository = 'https://github.com/truenas/charts.git'
+truechart_repository = 'https://github.com/ericbsd/charts.git'
+github_official_charts = 'https://api.github.com/repos/truenas/charts/contents/charts/'
+github_truechart_charts = 'https://api.github.com/repos/ericbsd/charts/contents/charts/'
+official_chart = []
+for chart_dict in GET(github_official_charts).json():
```

Comment:
We should make sure that we are only adding directories here. Github API tells us about that bit in the information it provides for each entry.

Reply:
I am not sure to understand,  the code takes the directory list from the GitHub API. The problem is the list contains dictionaries with all the information for each directory.
```
[
  {
    "name": "ipfs",
    "path": "charts/ipfs",
    "sha": "45481cc5c5d13cde02dbe7ddcd47fdb3790f4085",
    "size": 0,
    "url": "https://api.github.com/repos/truenas/charts/contents/charts/ipfs?ref=master",
    "html_url": "https://github.com/truenas/charts/tree/master/charts/ipfs",
    "git_url": "https://api.github.com/repos/truenas/charts/git/trees/45481cc5c5d13cde02dbe7ddcd47fdb3790f4085",
    "download_url": null,
    "type": "dir",
    "_links": {
      "self": "https://api.github.com/repos/truenas/charts/contents/charts/ipfs?ref=master",
      "git": "https://api.github.com/repos/truenas/charts/git/trees/45481cc5c5d13cde02dbe7ddcd47fdb3790f4085",
      "html": "https://github.com/truenas/charts/tree/master/charts/ipfs"
    }
  },
  {
    "name": "ix-chart",
    "path": "charts/ix-chart",
    "sha": "719158b0fe3ca02a715a7a13150d09a973e1e803",
    "size": 0,
    "url": "https://api.github.com/repos/truenas/charts/contents/charts/ix-chart?ref=master",
    "html_url": "https://github.com/truenas/charts/tree/master/charts/ix-chart",
    "git_url": "https://api.github.com/repos/truenas/charts/git/trees/719158b0fe3ca02a715a7a13150d09a973e1e803",
    "download_url": null,
    "type": "dir",
    "_links": {
      "self": "https://api.github.com/repos/truenas/charts/contents/charts/ix-chart?ref=master",
      "git": "https://api.github.com/repos/truenas/charts/git/trees/719158b0fe3ca02a715a7a13150d09a973e1e803",
      "html": "https://github.com/truenas/charts/tree/master/charts/ix-chart"
    }
  },
  {
    "name": "minio",
    "path": "charts/minio",
    "sha": "c38d301baa5293ae50ceb16f44eba44605a779ce",
    "size": 0,
    "url": "https://api.github.com/repos/truenas/charts/contents/charts/minio?ref=master",
    "html_url": "https://github.com/truenas/charts/tree/master/charts/minio",
    "git_url": "https://api.github.com/repos/truenas/charts/git/trees/c38d301baa5293ae50ceb16f44eba44605a779ce",
    "download_url": null,
    "type": "dir",
    "_links": {
      "self": "https://api.github.com/repos/truenas/charts/contents/charts/minio?ref=master",
      "git": "https://api.github.com/repos/truenas/charts/git/trees/c38d301baa5293ae50ceb16f44eba44605a779ce",
      "html": "https://github.com/truenas/charts/tree/master/charts/minio"
    }
  },
  {
    "name": "nextcloud",
    "path": "charts/nextcloud",
    "sha": "3f58ea9dc3b18196858b1a16606931a61acc0d30",
    "size": 0,
    "url": "https://api.github.com/repos/truenas/charts/contents/charts/nextcloud?ref=master",
    "html_url": "https://github.com/truenas/charts/tree/master/charts/nextcloud",
    "git_url": "https://api.github.com/repos/truenas/charts/git/trees/3f58ea9dc3b18196858b1a16606931a61acc0d30",
    "download_url": null,
    "type": "dir",
    "_links": {
      "self": "https://api.github.com/repos/truenas/charts/contents/charts/nextcloud?ref=master",
      "git": "https://api.github.com/repos/truenas/charts/git/trees/3f58ea9dc3b18196858b1a16606931a61acc0d30",
      "html": "https://github.com/truenas/charts/tree/master/charts/nextcloud"
    }
  },
  {
    "name": "plex",
    "path": "charts/plex",
    "sha": "33c3e0529c66683ca1790b2c0cccd5072c840576",
    "size": 0,
    "url": "https://api.github.com/repos/truenas/charts/contents/charts/plex?ref=master",
    "html_url": "https://github.com/truenas/charts/tree/master/charts/plex",
    "git_url": "https://api.github.com/repos/truenas/charts/git/trees/33c3e0529c66683ca1790b2c0cccd5072c840576",
    "download_url": null,
    "type": "dir",
    "_links": {
      "self": "https://api.github.com/repos/truenas/charts/contents/charts/plex?ref=master",
      "git": "https://api.github.com/repos/truenas/charts/git/trees/33c3e0529c66683ca1790b2c0cccd5072c840576",
      "html": "https://github.com/truenas/charts/tree/master/charts/plex"
    }
  }
]
```

Reply:
@ericbsd i was referring to filtering out these entries based on `type == dir` as only directories under this path are considered catalog items.

Reply:
Oh yeah, I got that after replying. I was to reply I got sidetrack.

Reply:
thanks anyway.

Reply:
No problem ;)